### PR TITLE
Add a formatter binary skeleton as P4C back end

### PR DESCRIPTION
**Pull request by snapdgn**
This is the initial setup to build a separate backend `p4fmt`, which will eventually function as a P4 code formatter.

Presently, it only outputs the source code prior to any frontend/midend/backend processing. It doesn't have any formatting capability as of now.
It is meant to demonstrate the working of the p4c's pretty-printer [toP4](https://github.com/p4lang/p4c/blob/main/frontends/p4/toP4/toP4.h).

### CODE REVIEW

File: backends/p4fmt/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+set(FMT_SRCS
+    p4fmt.cpp
+    options.cpp
+    )
```
Comment (fruffy) [inline backends/p4fmt/CMakeLists.txt:4]: We do not have a CMake style but I would fix this indentation here. 
```suggestion
)
```

File: backends/p4fmt/options.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    virtual ~P4fmtOptions() = default;
+    P4fmtOptions(const P4fmtOptions &) = default;
+    P4fmtOptions(P4fmtOptions &&) = delete;
+    P4fmtOptions &operator=(const P4fmtOptions &) = default;
+    P4fmtOptions &operator=(P4fmtOptions &&) = delete;
+
+    const cstring &outputFile() const;
+
+ private:
+    // File to output to.
```
Comment (fruffy) [inline backends/p4fmt/options.h:21]: Use three slashes here. 
```suggestion
    /// File to output to.
```

File: backends/p4fmt/p4fmt.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    if (options.process(argc, argv) == nullptr) {
+        return EXIT_FAILURE;
+    }
+
+    options.setInputFile();
+
+    std::ostream *out = nullptr;
+
+    // write to stdout in absence of an output file.
```
Comment (fruffy) [inline backends/p4fmt/p4fmt.cpp:26]:
```suggestion
    // Write to stdout in absence of an output file.
```

**Review by fruffy — APPROVED** [review]
Looks good, just some nit picks.

**Comment by snapdgn** [comment]
resolved nits & squashed the commits.

File: backends/p4fmt/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# p4fmt (P4 Formatter)
+
+p4fmt is a WIP formatter for P4.
```
Comment (fruffy) [inline backends/p4fmt/README.md:3]: I would some more detail here that this is not ready yet and not meant for use.

Comment (snapdgn) [reply backends/p4fmt/README.md]: Let me know if the updated one looks good!

**Review by fruffy — APPROVED** [review]

**Comment by fruffy** [comment]
Please rebase and fix the compilation errors, then we should be able to merge this. The problem is that the string needs to be initialized as cstring().

**Pull request merged**